sdk/internal/temporal/resource.go

```diff
@@ -121,3 +148,8 @@ func (er *Resource[TResource, TState]) Expire() {
 	// Reset the expiration as if we never got this resource to begin with
 	er.expiration = time.Time{}
 }
+
+func (r *Resource[TResource, TState]) expiringSoon(TResource, TState) bool {
```

Comment:
The receiver should be named `er` to match the other methods (worse name I know but then it's consistent; thought the linter would flag this).